File: DownloadLib/src/main/java/com/kronos/download/DownloadService.java
package com.kronos.download;

import android.app.IntentService;
import android.content.Intent;

/**
 * Created by Leif Zhang on 16/9/29.
 * Email leifzhanggithub@gmail.com
 */
public class DownloadService extends IntentService {

    public DownloadService() {
        super("DownloadService");
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        try {
            String url = intent.getStringExtra("url");
            DownloadManager.getInstance().startRequest(url);
            DownloadManager.getInstance().saveOne(url);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onDestroy() {
        try {
            DownloadManager.getInstance().saveAll();
        } catch (Exception e) {
            e.printStackTrace();
        }
        super.onDestroy();
    }
}

File: DownloadLib/src/main/java/com/kronos/download/DownloadService.kt
package com.kronos.download

import android.app.IntentService
import android.content.Intent

/**
 * Created by Leif Zhang on 16/9/29.
 * Email leifzhanggithub@gmail.com
 */
class DownloadService : IntentService("DownloadService") {

    override fun onHandleIntent(intent: Intent?) {
        try {
            val url = intent!!.getStringExtra("url")
            DownloadManager.startRequest(url)
            DownloadManager.saveOne(url)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun onDestroy() {
        try {
            DownloadManager.saveAll()
        } catch (e: Exception) {
            e.printStackTrace()
        }

        super.onDestroy()
    }
}
